lib/ansible/modules/system/ufw.py

```diff
@@ -245,10 +245,16 @@ def execute(cmd):
         cmd = ' '.join(map(itemgetter(-1), filter(itemgetter(0), cmd)))
```

Comment:
Ok, I really didn't expect the code to be *that* crazy. Could you replace this by
```suggestion
        cmd = list(map(itemgetter(-1), filter(itemgetter(0), cmd)))
```
? `module.run_command()` is totally capable of processing a list (instead of a string), and in fact, that's the WAY better method to call it. If you pass a string, that string first has to be parsed again (to be split as a list), and that list is then passed on...

Reply:
All right let's try :)

Reply:
I think this will change lot of the comportment of this function. There is commands with space mixed with command in list.

I haven't time to execute full regression test on this module (and there is no unit test). For a first version, and to enable check mode, we should keep the quote in the `grep '^### tuple'`

Reply:
True, you're right. It's definitely something we should clean up eventually, but for now let's keep it.